vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SearchManager.java

```diff
@@ -0,0 +1,270 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import edu.asu.diging.vspace.core.data.TextContentBlockRepository;
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.impl.Module;
+import edu.asu.diging.vspace.core.model.impl.Slide;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISearchManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.ISlideManager;
+import edu.asu.diging.vspace.core.services.impl.model.SearchModuleResults;
+import edu.asu.diging.vspace.core.services.impl.model.SearchSlideResults;
+import edu.asu.diging.vspace.core.services.impl.model.SearchSlideTextBlockResults;
+import edu.asu.diging.vspace.core.services.impl.model.SearchSpaceResults;
+
+
+public abstract class SearchManager implements ISearchManager {
+    
+    @Value("${page_size}")
+    private int pageSize;
+    
+    @Autowired
+    private ISequenceManager sequenceManager;
+    
+    protected abstract  Page<ISpace> spaceSearch(Pageable requestedPageForSpace ,  String searchTerm);
+    
+    protected abstract  Page<IModule> moduleSearch(Pageable requestedPageForModule ,  String searchTerm);
+    
+    protected abstract  Page<ISlide> slideSearch(Pageable requestedPageForSlide ,  String searchTerm);
+
+    protected abstract  Page<ISlide> slideTextSearch(Pageable requestedPageForSlideText ,  String searchTerm);   
+    
+    /**
+     * Method to return the requested spaces whose name or description contains the
+     * search string
+     * 
+     * @param searchTerm string which has been searched
+     * @param page.      The page number starts from 1.
+     * @return set of spaces whose name or description contains the search string in
+     *         the requested page.
+     */
+    @Override
+    public Page<ISpace> paginationInSpaces(String searchTerm, int page) {
```

Comment:
similar as above; name  should be an action that describes what this method is doing.